### PR TITLE
DAT-3994 | Infobox Builder - handle template naming conflict

### PR DESCRIPTION
**Pull request by dianafa**
DO NOT MERGE without APP PART: https://github.com/Wikia/app/pull/10071
(app part has to be released first)

## Links

https://wikia-inc.atlassian.net/browse/DAT-3994

## Description

Add template naming conflict handling. 

## Reviewers
@bognix @tomasznapieralski

### CODE REVIEW

File: front/main/app/models/infobox-builder.js [hunk 1]
```diff
@@ -298,6 +298,31 @@ const InfoboxBuilderModel = Ember.Object.extend({
 		state.forEach((element) => this.addItem(element.type, element));
 	},
 
+	getTemplateExists(title) {
+		return new Ember.RSVP.Promise((resolve, reject) => {
+			Ember.$.ajax({
```
Comment (bognix) [inline front/main/app/models/infobox-builder.js]: how about using $.getJSON?
http://api.jquery.com/jquery.getjson/

File: front/main/tests/unit/components/infobox-builder-test.js [hunk 2]
```diff
@@ -303,48 +303,6 @@ test('tryGoToSource handles opening of go to source modal', function (assert) {
 	});
 });
 
-test('tryGoToSource handles edit title modal used', function (assert) {
```
Comment (bognix) [inline front/main/tests/unit/components/infobox-builder-test.js]: why did you remove the tests? :(

**Comment by bognix** [comment]
minor notes - LGTM

File: front/main/tests/unit/components/infobox-builder-test.js (hunk 2 above)
Comment (dianafa) [inline front/main/tests/unit/components/infobox-builder-test.js]: tests are testng `component.get('canGoToSourceModal')` and this property has been removed

**Comment by tomasznapieralski** [comment]
LGTM
